lib/extract/netcdf/Makefile

```diff
@@ -32,42 +32,49 @@
 # POSSIBILITY OF SUCH DAMAGE.
 # -----------------------------------------------------------------------------
 # Author: J. Henrichs, Australian Bureau of Meteorology
+# Modified: I. Kavcic, Met Office
 
 # This Makefile just creates the Fortran base class for all extraction
 # libraries from the jinja template. The compile functionality it offers is
 # only meant for testing, each library should create and compile its own
 # version of the base class (with therefore consistent compiler and
 # jinja settings).
 
+# ----------- Default "make" values, can be overwritten by the user -----------
+# Compiler and compiler flags
 F90 ?= gfortran
-F90FLAGS ?= 
+F90FLAGS ?=
+# Path to the PSyclone wrapper libraries. It defaults to the relative path to
+# the top-level 'lib' directory. Overwrite for a custom location.
+PSYDATA_LIB_DIR ?= ./../..
+# -----------------------------------------------------------------------------
 
 F90FLAGS += $$(nf-config --fflags)
 
 # The extract library is implemented for int, real and
 # double scalars and 2-dimension arrays
 PROCESS_ARGS = -prefix=extract_ -types=int,real,double \
 		       -dims=2
-PROCESS=$$(../../get_python.sh) ../../process.py
+PROCESS = $$($(PSYDATA_LIB_DIR)/get_python.sh) $(PSYDATA_LIB_DIR)/process.py
 
 default: extract_netcdf_base.o psy_data_base.o
+
 process: extract_netcdf_base.f90
 
 .PHONY: default process clean all
 
 all:
-	$(MAKE) -C dl_esm_inf
-	$(MAKE) -C lfric
-
+	$(MAKE) -C dl_esm_inf F90="$(F90)" F90FLAGS="$(F90FLAGS)"
```

Comment:
I removed the flags as they are passed to the library compilation. I added them as I thought that would force the libraries to be recompiled if the compiler is changed, however it doesn't work like that. I added the ``allclean`` target here to remove the created ``dl_esm_inf`` and ``lfric`` infrastructure libraries that are created with ``make all``.

Reply:
Another remark: I noticed this `Makefile` and the `lib/read_only/Makefile` have `all` target that compiles `dl_esm_inf` and `lfric` libraries in the respective subdirectories. `lib/nan_test/Makefile`, however, does not. What is the logic behind adding these targets? Anyway, I went by the logic that if these targets are present I add the `allclean` target to clean up the created infrastructure files to not mess up the compilation when switching compiler environments.

Reply:
I've opened #1209 so I can have a look at these inconsistencies, they are not really related to this PR, so I am happy to fix it all separately, especially since I seem to have the problems with the missing NetCDF flags when passing the variables explicitly.

Reply:
I believe you are talking of the (phony) targets that should trigger compilation of the infrastructure libraries when they do not exist?

Reply:
Yes, this is what I was talking about. I was confusing targets and flags, it seems...